src/batSplay.ml

```diff
@@ -498,7 +514,7 @@ struct
     let rec bfind = function
       | Node (Empty, kv, r) -> mini := kv; r
       | Node (l, kv, r) -> Node (bfind l, kv, r)
-      | Empty -> assert(false)
+      | Empty -> raise Not_found
```

Comment:
Consistency with max_binding, choose, find (_first, _last), Consistency with BatMap and compatibility with the stdlib: https://ocaml.org/releases/4.11/htmlman/libref/Map.S.html

Note that the batteries documentation does not (yet) specify which exception is raised (which, again, should be fixed by this PR), see http://ocaml-batteries-team.github.io/batteries-included/hdoc2/BatMap.html

Reply:
I mean, the current situation for batMap is:
- min_binding and any raise Not_found
- pop_min_binding and pop_max_binding have assert(false)
- max_binding has invalid_arg "PMap.max_binding: empty tree"
  (the error message seems... outdated?)
None of these exceptions are documented.

For BatSplay the current situation is:
- min_binding, any and max_binding raise Not_found
- pop_min_binding and pop_max_binding have assert(false)

Reply:
The authors of this code used `assert false` because they believe that this point can never be reached, for any input. I haven't looked at the code deeply, but I think that the reasoning is that `choose tr` was called first, so we already know that there is at least one element. If you find this unclear, I think it would be a good idea to add a comment. If this is wrong for some reason I missed, then indeed it needs to be changed. But if it is true that this case can never be reached, then I think `assert false` is the right choice.

Reply:
You're right, i missed that get_root (for batMap) and choose (for batSplay) are called before, and raise Not_found - so the assert(false) here should be unreachable.

That means that the only place where the "wrong" (in my opinion) exception is raised is in max_binding, which does not raise `Not_found`.

I will reinstate the `assert(false)` and add a comment.

Reply:
Thanks! (I agree that the exception for `max_binding` was wrong.)